rct229/rulesets/ashrae9012019/section19/section19rule10.py

```diff
@@ -0,0 +1,271 @@
+    HVAC_SYS.SYS_12,
+    HVAC_SYS.SYS_13,
+]
+
+SYSTEM_3_4_TYPES = [HVAC_SYS.SYS_3, HVAC_SYS.SYS_4]
+
+
+class Section19Rule10(RuleDefinitionListIndexedBase):
+    """Rule 10 of ASHRAE 90.1-2019 Appendix G Section 19 (HVAC - General)"""
+
+    def __init__(self):
+        super(Section19Rule10, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, True),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section19Rule10.HVACRule(),
+            index_rmr="baseline",
+            id="19-10",
+            description="Air economizers shall be included in baseline HVAC Systems 3 through 8, and 11, 12, and 13 based on climate as specified in Section G3.1.2.6 with exceptions."
+            "1. Systems that include gas-phase air cleaning to meet the requirements of Standard 62.1, Section 6.1.2. This exception shall be used only if the system in the proposed design does not match the building design."
+            "2. Where the use of outdoor air for cooling will affect supermarket open refrigerated case-work systems. This exception shall only be used if the system in the proposed design does not use an economizer. If the exception is used, an economizer shall not be included in the baseline building design."
+            "3. Systems that serve computer rooms complying with Section G3.1.2.6.1.",
+            ruleset_section_title="HVAC - General",
+            standard_section="Section G3.1.2.6 including exceptions 1-3",
+            is_primary_rule=True,
+            list_path="$.ruleset_model_instances[*].buildings[*].building_segments[*].heating_ventilating_air_conditioning_systems[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmr_b = context.baseline
+
+        assert_(
+            rmr_b["ruleset_model_instances"],
+            "Please make sure `ruleset_model_instances` exists in the `baseline`.",
+        )
+        rmi_b = rmr_b["ruleset_model_instances"]
+
+        climate_zone = rmr_b["weather"]["climate_zone"]
+
+        baseline_system_types_dict_b = get_baseline_system_types(rmi_b)
+
+        return (climate_zone not in NOT_APPLICABLE_CLIMATE_ZONE) and any(
+            [
+                baseline_system_type_compare(system_type, applicable_sys_type, False)
+                for system_type in baseline_system_types_dict_b.keys()
+                for applicable_sys_type in APPLICABLE_SYS_TYPES
+            ]
+        )
+
+    def create_data(self, context, data):
+        rmr_b = context.baseline
+        rmi_b = rmr_b["ruleset_model_instances"][0]
+
+        rmr_p = context.proposed
+        assert_(
+            rmr_p["ruleset_model_instances"],
+            "Please make sure `ruleset_model_instances` exists in the `proposed`.",
+        )
+        rmi_p = rmr_p["ruleset_model_instances"][0]
+
+        proposed_has_economizer = is_economizer_modeled_in_proposed(rmi_b, rmi_p)
```

Comment:
This is incorrect. The `is_economizer_modeled_in_proposed` suppose to take `rmi_b`, `rmi_p` and `hvac_id_b` as argument.
It is testing whether the proposed HVAC systems associated with zones that are connected with `hvac_id_b` have economizers.
This flag should move to the `hvac_system_exception_2_list`

I updated to the `is_economizer_modeled_in_proposed` function

Reply:
Addressed.